ArduCopter/Copter.cpp

```diff
@@ -548,11 +559,11 @@ void Copter::ten_hz_logging_loop()
 // twentyfive_hz_logging - should be run at 25hz
 void Copter::twentyfive_hz_logging()
 {
-    if (should_log(MASK_LOG_ATTITUDE_FAST)) {
+    if (should_log(MASK_LOG_ATTITUDE_FAST) && (!should_log(MASK_LOG_ATTITUDE_MED))) {
```

Comment:
I think this is the fast logging, so we should not disable fast logging if medium is enabled. It should be the other way, medium should be disabled if doing fast.

Reply:
FAST logs ATT at loop, POS at 25....MED log ATT at 10 and POS at 10....this in no change, only making FAST override MED in both log types...fastest wins

Reply:
In this if medium is overriding fast?

Reply:
no POS with ATT_FAST was 25hz, with MED it was 10hz....still is

Reply:
This line should not change
```suggestion
    if (should_log(MASK_LOG_ATTITUDE_FAST)) {
```
To have the correct change to stop double logging on line 516 of the diff.

With this line as it is you log at 10hz if both fast and medium are set, in that case we should log at 25hz which is what the code does without the change.

Reply:
I really dont....on line 516 its an OR not an AND ....I need to change 516 to  MED && !FAST || !MED && !FAST ...the intent was to log POS even if ATT was not logged...a valid scenario I think...then the change should be removed as you state in 559....do you agree? FAST checked logs at 25H POS, MED and not FAST logs at 10HZ, and neither checked logs at 10HZ